test/testdata/infer/all_bug.rb

```diff
@@ -18,6 +18,7 @@ def self.test(x)
   end
 end
 
-T.assert_type!(Test.test(T.untyped), M)
+T.assert_type!(Test.test(T.untyped), M) # error: This code is unreachable
+                       # ^^^^^^^^^ error: Expected `T.all(M, T)` but found `<Type: T.untyped>` for argument `x`
```

Comment:
My guess is that the author's original intention with this bug was to use `T.unsafe(nil)` here.

Reply:
i.e., we should hope that we didn't change the behavior of the `This code is unreachable error` because we didn't change `T.all`